tracing-subscriber/src/registry/mod.rs

```diff
@@ -45,3 +54,96 @@ pub trait LookupMetadata {
         self.metadata(id).is_some()
     }
 }
+
+pub trait LookupSpan<'a> {
+    type Data: SpanData<'a>;
+    fn span_data(&'a self, id: &Id) -> Option<Self::Data>;
+
+    fn span(&'a self, id: &Id) -> Option<SpanRef<'_, Self>>
+    where
+        Self: Sized,
+    {
+        let data = self.span_data(id)?;
+        Some(SpanRef {
+            registry: self,
+            data,
+        })
+    }
+}
+
+pub trait SpanData<'a> {
+    type Follows: Iterator<Item = &'a Id>;
+
+    fn id(&self) -> Id;
+    fn metadata(&self) -> &'static Metadata<'static>;
+    fn parent(&self) -> Option<&Id>;
+    fn follows_from(&'a self) -> Self::Follows;
+    fn extensions(&self) -> Extensions<'_>;
+    fn extensions_mut(&self) -> ExtensionsMut<'_>;
+}
```

Comment:
these new traits need comments; since I wrote them, I would be happy to add the comments...